db/background_mgr_attachment_migration_test.go

```diff
@@ -0,0 +1,312 @@
+	assert.Equal(t, int64(0), stats.DocsChanged)
+
+	// assert that the sync info metadata version doc has been written to the database collection
+	var syncInfo base.SyncInfo
+	_, err = collection.dataStore.Get(base.SGSyncInfo, &syncInfo)
+	require.NoError(t, err)
+	assert.Equal(t, int8(4), syncInfo.MetaDataVersion)
+}
+
+func TestMigrationManagerDocWithSyncAndGlobalAttachmentMetadata(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("rosmar does not support DCP client, pending CBG-4249")
+	}
+	db, ctx := setupTestDB(t)
+	defer db.Close(ctx)
+	collection, ctx := GetSingleDatabaseCollectionWithUser(ctx, t, db)
+
+	docBody := Body{
+		"value":         1234,
+		BodyAttachments: map[string]interface{}{"myatt": map[string]interface{}{"content_type": "text/plain", "data": "SGVsbG8gV29ybGQh"}},
+	}
+	key := t.Name()
+	_, _, err := collection.Put(ctx, key, docBody)
+	require.NoError(t, err)
+
+	xattrs, cas, err := collection.dataStore.GetXattrs(ctx, key, []string{base.SyncXattrName, base.GlobalXattrName})
+	require.NoError(t, err)
+	require.NotNil(t, xattrs[base.GlobalXattrName])
+	require.NotNil(t, xattrs[base.SyncXattrName])
+
+	var syncData SyncData
+	require.NoError(t, base.JSONUnmarshal(xattrs[base.SyncXattrName], &syncData))
+	// define some attachment meta on sync data
+	syncData.Attachments = AttachmentsMeta{}
+	att := map[string]interface{}{
+		"stub": true,
+	}
+	syncData.Attachments["someAtt.txt"] = att
+
+	updateXattrs := map[string][]byte{
+		base.SyncXattrName: base.MustJSONMarshal(t, syncData),
+	}
+	_, err = collection.dataStore.UpdateXattrs(ctx, key, 0, cas, updateXattrs, DefaultMutateInOpts())
+	require.NoError(t, err)
+
+	attachMigrationMgr := NewAttachmentMigrationManager(db.MetadataStore, db.MetadataKeys)
+	require.NotNil(t, attachMigrationMgr)
+
+	options := map[string]interface{}{
+		"database": db,
+	}
+	err = attachMigrationMgr.Start(ctx, options)
+	require.NoError(t, err)
+
+	// wait for task to complete
+	err = WaitForConditionWithOptions(t, func() bool {
+		var status BackgroundManagerStatus
+		rawStatus, _ := attachMigrationMgr.GetStatus(ctx)
+		_ = base.JSONUnmarshal(rawStatus, &status)
+		return status.State == BackgroundProcessStateCompleted
+	}, 200, 200)
+	require.NoError(t, err)
+
+	// assert that the two added docs above were processed but not changed
+	stats := getAttachmentMigrationStats(attachMigrationMgr.Process)
+	assert.Equal(t, int64(1), stats.DocsProcessed)
+	assert.Equal(t, int64(1), stats.DocsChanged)
+
+	// assert that the sync info metadata version doc has been written to the database collection
+	var syncInfo base.SyncInfo
+	_, err = collection.dataStore.Get(base.SGSyncInfo, &syncInfo)
+	require.NoError(t, err)
+	assert.Equal(t, int8(4), syncInfo.MetaDataVersion)
+
+	xattrs, _, err = collection.dataStore.GetXattrs(ctx, key, []string{base.SyncXattrName, base.GlobalXattrName})
+	require.NoError(t, err)
+	require.NotNil(t, xattrs[base.GlobalXattrName])
+	require.NotNil(t, xattrs[base.SyncXattrName])
```

Comment:
Using contains will log what xattrs are included.

```suggestion
	require.Contains(t, xattrs, base.GlobalXattrName)
	require.Contains(t, xattrs, base.SyncXattrName)
```